### PR TITLE
Feature/54082

### PR DESCRIPTION
**Pull request by bartdominiak**

### CODE REVIEW

File: Snowdog_SolrNavigation/templates/grid.phtml [hunk 1]
```diff
@@ -141,3 +147,11 @@ $solrDataHelper = $this->helper('Snowdog\SolrNavigation\Helper\Data');
...
     <?php endif; ?>
 <?php endif; ?>
+
+<script type="text/x-magento-init">
+    {
+        "[data-role=tocart-form], .form.map.checkout": {
+            "solrAddToCart": {}
+        }
+    }
+</script>
```
Comment (szafran89) [inline Snowdog_SolrNavigation/templates/grid.phtml]: Can we change to `data-mage-init`?

File: Snowdog_SolrNavigation/requirejs-config.js [hunk 2]
```diff
@@ -4,7 +4,8 @@ var config = {
       'priceRange': 'Snowdog_SolrNavigation/js/price-range.babel',
       'quickSearch': 'Snowdog_SolrNavigation/js/quicksearch.babel',
       'range': 'Snowdog_SolrNavigation/js/range.babel',
-      'toolbar': 'Snowdog_SolrNavigation/js/toolbar.babel'
+      'toolbar': 'Snowdog_SolrNavigation/js/toolbar.babel',
+      'solrAddToCart': 'Snowdog_SolrNavigation/js/solr-add-to-cart'
```
Comment (szafran89) [inline Snowdog_SolrNavigation/requirejs-config.js]: Please update chengelog

File: Snowdog_SolrNavigation/templates/list.phtml [hunk 3]
```diff
@@ -130,3 +138,11 @@ $solrDataHelper = $this->helper('Snowdog\SolrNavigation\Helper\Data');
         <?php endforeach; ?>
     <?php endif; ?>
 <?php endif; ?>
```
Comment (szafran89) [inline Snowdog_SolrNavigation/templates/list.phtml]: remove semicolons

File: Snowdog_SolrNavigation/templates/list.phtml [hunk 4]
```diff
@@ -130,3 +138,11 @@ $solrDataHelper = $this->helper('Snowdog\SolrNavigation\Helper\Data');
...
+
+<script type="text/x-magento-init">
+    {
+        "[data-role=tocart-form], .form.map.checkout": {
+            "solrAddToCart": {}
+        }
+    }
+</script>
```
Comment (szafran89) [inline Snowdog_SolrNavigation/templates/list.phtml]: Can we change to data-mage-init?

File: Snowdog_SolrNavigation/web/js/solr-add-to-cart.js [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+      addToCartButton.attr('title', addToCartButtonTextAdded);
+
+      setTimeout(function () {
+        var addToCartButtonTextDefault = $t('Add to Cart');
+
+        addToCartButton.prop('disabled', false);
+        addToCartButton.removeClass('disabled');
+        addToCartButton.find('span').text(addToCartButtonTextDefault);
+        addToCartButton.attr('title', addToCartButtonTextDefault);
+      }, 2500);
```
Comment (szafran89) [inline Snowdog_SolrNavigation/web/js/solr-add-to-cart.js]: why 2500?

File: Snowdog_SolrNavigation/web/js/solr-add-to-cart.js [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+    }
+  });
+
+  return $.snowdog.catalogAddToCart;
+});
```
Comment (szafran89) [inline Snowdog_SolrNavigation/web/js/solr-add-to-cart.js]: missing new line

File: Snowdog_SolrNavigation/templates/list.phtml [hunk 7]
```diff
@@ -86,7 +92,9 @@ $solrDataHelper = $this->helper('Snowdog\SolrNavigation\Helper\Data');
                                     "
                                     type="submit"
                                 >
-                                    <?= __('Add to Cart') ?>
+                                    <span>
+                                        <?= __('Add to Cart') ?>
+                                    </span>
```
Comment (szafran89) [inline Snowdog_SolrNavigation/templates/list.phtml]: why did you add span? Can we remove that?

File: Snowdog_SolrNavigation/templates/grid.phtml (hunk 1 above)
Comment (bartdominiak) [reply Snowdog_SolrNavigation/templates/grid.phtml]: Sure. Added `data-mage-init`

File: Snowdog_SolrNavigation/web/js/solr-add-to-cart.js (hunk 5 above)
Comment (bartdominiak) [reply Snowdog_SolrNavigation/web/js/solr-add-to-cart.js]: For two reasons:
- Protect server overload (You can spam click on active button and make a lot of unecessary ajax request)
- UX - IMO User should receive a longer message b/c right it's chaning like:
* "Adding" - 1 second
* "Added" - 1 second (increased to 2.5second)

File: Snowdog_SolrNavigation/templates/list.phtml (hunk 7 above)
Comment (bartdominiak) [reply Snowdog_SolrNavigation/templates/list.phtml]: b/c of `solrAddToCart`: 

```js
addToCartButton.find('span').text(addToCartButtonTextDefault);
```
